File: Juego/Mago.java
package Juego;

public class Mago extends Personaje {
    private int mana;

    public Mago(String nombre, int puntosDeVida, int mana) {
        super(nombre, puntosDeVida);
        this.mana = mana;
    }

    public void lanzarHechizo(Hechizo hechizo, Personaje objetivo) {
        if (mana >= hechizo.getCostoMana()) {
            mana -= hechizo.getCostoMana();
            System.out.println(getNombre() + " lanza el hechizo " + hechizo.getNombre() + " contra " + objetivo.getNombre() + ".");
            objetivo.recibirDaño(hechizo.getDaño());
        } else {
            System.out.println(getNombre() + " no tiene suficiente maná para lanzar " + hechizo.getNombre() + ".");
        }
    }


    public void atacar(Personaje objetivo) {
        System.out.println(getNombre() + " ataca a " + objetivo.getNombre() + " con un bastón.");
        objetivo.recibirDaño(5);
    }
}

File: Juego/Mago.kt
package Juego

class Mago(nombre: String?, puntosDeVida: Int, private var mana: Int) : Personaje(nombre!!, puntosDeVida) {
    fun lanzarHechizo(hechizo: Hechizo, objetivo: Personaje) {
        if (mana >= hechizo.costoMana) {
            mana -= hechizo.costoMana
            println(nombre + " lanza el hechizo " + hechizo.nombre + " contra " + objetivo.nombre + ".")
            objetivo.recibirDaño(hechizo.daño)
        } else {
            println(nombre + " no tiene suficiente maná para lanzar " + hechizo.nombre + ".")
        }
    }


    override fun atacar(objetivo: Personaje?) {
        println(nombre + " ataca a " + objetivo?.nombre + " con un bastón.")
        objetivo?.recibirDaño(5)
    }
}